core/tx_pool.go

```diff
@@ -336,7 +336,8 @@ func NewTxPool(config TxPoolConfig, chainconfig *params.ChainConfig, chain block
 	pool.reset(nil, chain.CurrentBlock().Header())
 	// TODO: Does this reordering cause a problem?
 	// priced list depends on the current ctx which is set in reset
-	pool.priced = newTxPricedList(pool.all, &pool.currentCtx)
+	// Use the global slots as the max amount of stale transactions in the priced heap before a re-heap.
+	pool.priced = newTxPricedList(pool.all, &pool.currentCtx, int64(pool.config.GlobalSlots))
```

Comment:
i guess logic here is that we can have as many stales a tx have in the pool, if we consider that a tx usually is 1 slot, which i'm not sure if true in average.

Still, it's works a default number, just a bit ackward

Reply:
if stales > as globalslots and a reheap hadn't been done before, it means that the heaps sizes are at a minimum 4x the size (due to the 25% rule) so something is off. A lower number could be set, but I feel like as a ceiling it should be ok.